riscv/mmu.h

```diff
@@ -290,8 +290,10 @@ class mmu_t
     if (matched_trigger)
       throw *matched_trigger;
 
+    reg_t vpn = addr >> PGSHIFT;
     auto access_info = generate_access_info(addr, FETCH, {false, false, false});
-    check_triggers(triggers::OPERATION_EXECUTE, addr, access_info.effective_virt);
+    if (unlikely(tlb_insn_tag[vpn % TLB_ENTRIES] & TLB_CHECK_TRIGGERS))
```

Comment:
This doesn't look right to me.  What if this is a TLB conflict miss and some unrelated entry (with the same VPN LSBs) is in that entry?

The old design where the call to `check_triggers` lived in `fetch_slow_path` made more sense to me.  We knew we were guaranteed to end up in `fetch_slow_path` if `TLB_CHECK_TRIGGERS` was true, because the comparison in `translate_insn_addr` would never match.

Reply:
Update the commit according to @aswaterman's suggestion.